### PR TITLE
#184581177 authorization: protected routes (all) should only be accessed by autheticated users

### PR DESCRIPTION
**Pull request by 1MuhireDavid**
#### What does this PR do?
This PR adds authorization to protected routes so that they can be accessed by authorized uses.

#### Description of Task to be completed?
- Add a middleware to authorize protected routes according to their role 

#### How should this be manually tested?
- Clone this repo git clone https://github.com/atlp-rwanda/e-comm-team-amigos-bn
- Switch your working directory cd e-comm-team-amigos-bn
- Create an .env file and update it with .env-example
- Install all the dependencies by running npm install
- Then run the project by typing npm start

#### Endpoints to be tested:
- `user/check`

#### Any background context you want to provide?
- Have Node Js installed in your local machine

#### What are the relevant pivotal tracker/Trello stories?

[#184581177](https://www.pivotaltracker.com/n/projects/2632829/stories/184581177)

#### Screenshots (if appropriate)
Screen Shot 2023-03-16 at 16 18 25 image for test

### CODE REVIEW

**Review by ericndungutse — APPROVED** [review]
LGTM

File: src/controllers/authentication.js [hunk 1]
```diff
@@ -66,6 +65,10 @@ export const loginUser = async (req, res) => {
   if (!user) {
     return res.status(400).json({ message: 'Email or Password Incorrect' });
   }
+  if (!user) { return res.status(400).json({ message: 'Email or Password Incorrect' }); }
+  // const valid = await bcrypt.compare(req.body.password, user.password);
```
Comment (nshutijonathan) [inline src/controllers/authentication.js:69]: remove commented out codes from the repo

File: src/controllers/authentication.js [hunk 2]
```diff
@@ -66,6 +65,10 @@ export const loginUser = async (req, res) => {
...
+  if (!user) { return res.status(400).json({ message: 'Email or Password Incorrect' }); }
+  // const valid = await bcrypt.compare(req.body.password, user.password);
```
Comment (nshutijonathan) [inline src/controllers/authentication.js:71]: remove commented out codes from the repo

File: src/database/seeders/20230302105921-User.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import { v4 as uuidv4 } from "uuid";
 import bcrypt from "bcryptjs";
 module.exports = {
+<<<<<<< HEAD
```
Comment (nshutijonathan) [inline src/database/seeders/20230302105921-User.js:4]: you did not rebase correctly  `<<<<<<< HEAD`

File: src/database/seeders/20230302105921-User.js [hunk 4]
```diff
@@ -44,4 +45,32 @@ module.exports = {
...
+        verified: 'true',
+        createdAt: new Date(),
+        updatedAt: new Date(),
+      }
+    ],
+    {},
+  ),
+  down: (queryInterface, Sequelize) => queryInterface.bulkDelete('Users', null, {}),
+};
+<<<<<<< HEAD
```
Comment (nshutijonathan) [inline src/database/seeders/20230302105921-User.js:73]: you did not rebase correctly <<<<<<< HEAD

File: src/database/seeders/20230302105921-User.js [hunk 5]
```diff
@@ -44,4 +45,32 @@ module.exports = {
...
+>>>>>>> ch(user): add authorization
+=======
+>>>>>>> c7571d3889fc32488bff605098720d2d3e549411
```
Comment (nshutijonathan) [inline src/database/seeders/20230302105921-User.js:76]: you did not rebase correctly <<<<<<< HEAD

**Review by nshutijonathan — CHANGES_REQUESTED** [review]
Check some comments and narrow down your commit messages to one two

**Pull request closed without merging**